### ISSUE TITLE
Incorrect PowerShell version with PowerShell-Core

### ISSUE THREAD

**Issue by IzaacJ** [kiedtl/winfetch#99]
PowerShell: PowerShell-Core 7.1.3.20210315 (Chocolatey)
winfetch: 2.2.0 (Chocolatey)

winfetch incorrectly lists my PowerShell version as 5.1.19041.1023. Manually running `$PSVersionTable.PSVersion`, directly afterwards, in the same terminal, correctly displays version 7.1.3.

```
PS C:\Users\izaac> winfetch

                     ....,,:;+ccllll   izaac@IZAAC-DESKTOP
       ...,,+:;  cllllllllllllllllll   -------------------
 ,cclllllllllll  lllllllllllllllllll   OS: Windows 10 Pro [64 bitar]
 llllllllllllll  lllllllllllllllllll   Host: To be filled by O.E.M. To be filled by O.E.M.
 llllllllllllll  lllllllllllllllllll   Kernel: 10.0.19043.0
 llllllllllllll  lllllllllllllllllll   Motherboard: ASUSTeK COMPUTER INC. CROSSHAIR V FORMULA-Z
 llllllllllllll  lllllllllllllllllll   Uptime: 22 hours 25 minutes
 llllllllllllll  lllllllllllllllllll   Packages: 17 (choco)
                                       Shell: PowerShell v5.1.19041.1023
 llllllllllllll  lllllllllllllllllll   Resolution: 1920x1080, 1920x1080
 llllllllllllll  lllllllllllllllllll   Terminal: Windows Terminal
 llllllllllllll  lllllllllllllllllll   CPU: AMD FX(tm)-8350 Eight-Core Processor
 llllllllllllll  lllllllllllllllllll   GPU: Radeon RX 580 Series
 llllllllllllll  lllllllllllllllllll   Memory: 7.73 GiB / 15.9 GiB (48%)
 `'ccllllllllll  lllllllllllllllllll   Disk (C:): 152 GiB / 446 GiB (34%)
       `' \\*::  :ccllllllllllllllll
                        ````''*::cll
                                  ``

PS C:\Users\izaac> $PSVersionTable

Name                           Value
----                           -----
PSVersion                      7.1.3
PSEdition                      Core
GitCommitId                    7.1.3
OS                             Microsoft Windows 10.0.19043
Platform                       Win32NT
PSCompatibleVersions           {1.0, 2.0, 3.0, 4.0…}
PSRemotingProtocolVersion      2.3
SerializationVersion           1.1.0.1
WSManStackVersion              3.0

PS C:\Users\izaac> $PSVersionTable.PSVersion

Major  Minor  Patch  PreReleaseLabel BuildLabel
-----  -----  -----  --------------- ----------
7      1      3
```

**Comment by rashil2000** [comment]
I haven't used chocolatey that much, but I think the problem is that it is executing the winfetch.ps1 script using system powershell. Can you go to the script's install location and execute it directly under powershell core to see if it's correct?

**Comment by bilditup1** [comment]
The batch file at `C:\ProgramData\chocolatey\bin` invokes `powershell`, which is what the system PS binary is called. Changing this to `pwsh` will have it run under Powershell 7+ instead. The code doesn't look like it's in the repo (do you guys maintain that distro yourselves?) but maybe if the script were changed to something like this, it could solve it?

```
@echo off
SET _ps_cmd=pwsh
where pwsh > nul
IF %errorlevel% NEQ 0 SET _ps_cmd=powershell
%_ps_cmd% -NoProfile -ExecutionPolicy unrestricted -Command "& 'C:\ProgramData\chocolatey\lib\winfetch\tools\winfetch-2.0.0\winfetch.ps1'  %*"
```

**Comment by rashil2000** [comment]
The chocolatey package is maintained by our friend @jakublevy.

@jakublevy is there a way to rectify this issue?

**Comment by jakublevy** [comment]
(Sorry for the delay, I missed this issue)

The described behavior is consistent with [Install-ChocolateyPowershellCommand](https://docs.chocolatey.org/en-us/create/functions/install-chocolateypowershellcommand#mainContent) which is used in winfetch Chocolatey installation.


Install-ChocolateyPowershellCommand always creates this batch file
```
@echo off
powershell -NoProfile -ExecutionPolicy unrestricted -Command "& 'C:\ProgramData\chocolatey\lib\winfetch\tools\winfetch.ps1'  %*"
````
So basically when calling `winfetch` from Powershell Core, you execute `winfetch.bat` which executes `winfetch.ps1` from a new Powershell host.

This chocolatey command (although recommended on Chocolatey docs) is actually not really robust and there is no way to use it with Powershell Core.

So, I will rewrite the installer without Install-ChocolateyPowershellCommand so that when `winfetch` is executed from:
- Powershell Core host, it will be executed by Powershell Core
- Powershell host, it will be executed by Powershell
- cmd, it will check whether Powershell Core is available and execute winfetch from it, otherwise, Powershell will be used.

The installer on Chocolatey will be probably updated with the next winfetch release (depends on Chocolatey moderation speed and my personal time).

**Comment by rashil2000** [comment]
Yes, that'd be pretty nice!

**Comment by bilditup1** [comment]
Fascinating!

**Comment by jakublevy** [comment]
I rewrote the installation and re-pushed version 2.2.0 to Chocolatey. I could do this because version 2.2.0 is still not moderated by Chocolatey (even 2.1.0 is still not moderated ⬤).

In addition to what I promised in my previous post, there is one new feature - tab completion for `winfetch` arguments in Powershell and Powershell Core.

The installer works differently now, `winfetch.ps1` is now in Chocolatey binary directory (this enables tab completion and execution in the current Powershell host).

There is also a simple `winfetch.bat` file in Chocolatey binary directory that is for `cmd` users. It checks if `pwsh` is present and uses it, otherwise, it uses `powershell` for execution.

In case of calling `winfetch.bat` from `pwsh` or `powershell` (why would anyone do this?), there is a simple (not really robust, but fast ▲) check whether running from `powershell` or `pwsh`, it uses the determined host for execution.

▲ [There are ways to do this reliably](https://stackoverflow.com/questions/53447286/in-a-cmd-batch-file-can-i-determine-if-it-was-run-from-powershell), but they require executing powershell script that adds ~1 second overhead, this would result in much slower execution of `winfetch.bat` from `cmd`, so I refrained from doing so.

⬤ See #101

**Comment by rashil2000** [comment]
Regarding ▲:
PowerShell script files have a higher priority inside PowerShell sessions (higher than PATHEXT members), so I don't even think there's a way to call `winfetch.bat` from PowerShell if you already have `winfetch.ps1` in your PATH (unless of course you call the filename by specifying its extension too). I think this is a really niche use case, so let's not worry about it too much.

**Comment by jakublevy** [comment]
The problem is already solved and pushed to Chocolatey package of winfetch 2.2.0.
Since this version is not moderated (and won't be sooner than winfetch becomes a trusted package at which point no moderation would be required), this issue should be closed with #101.
